texar/modules/encoders/rnn_encoders.py

```diff
@@ -460,6 +461,17 @@ def output_layer(self) -> Optional[nn.Module]:
         """
         return self._output_layer
 
+    @property
+    def output_size(self) -> int:
+        r"""The output size.
+        """
+        dim = self._cell.hidden_size
+        dummy_tensor = torch.Tensor(dim)
+        if self._output_layer is not None:
+            return self._output_layer(dummy_tensor).size()[-1]
```

Comment:
Use `.size(-1)`. And please add a TODO here saying we'll change the implementation to something that does not require a forward pass.